### PR TITLE
[3.x] com_wrapper. Add HTML5 mode without B\C break.

### PR DESCRIPTION
**Pull request by ReLater**
Related to https://github.com/joomla/joomla-cms/pull/19965
See discussion there concerning B\C breaks in Joomla 3.x.

@BaleshSrle
**Please feel free to take over code and/or idea into your pr. I have no problems to close this one then.**

### Summary of Changes
- Adds a new switch `Legacy (HTML4)` versus `HTML5`.
- `Legacy` is the default setting. Thus, this pr will not break older template overrides that use HTML4 attributes for IFRAME tags that have been removed in HTML5.
- Doesn't touch old parameter types and names but translates them via PHP for `HTML5` mode if necessary.
- `HTML5` uses CSS style instead. 
- Comment: Because most current browsers don't support CSS rule `overflow` for displaying/hiding scroll bars `scrolling` attribute cannot be removed in HTML5 mode.

### Testing Instructions
- Apply patch.
- Create a menu item of type "Wrapper" with setting `Mode = Legacy`.
- Create another menu item of type "Wrapper" with setting `Mode = HTML5`.

- Compare outputs in frontend.
- Test other settings and compare output in frontend.
- As far as I see: Old template overrides should still work even if you switch mode to `HTML5` because only default.php of core respects new setting. 

### Expected result
- No optical difference
 
### Documentation Changes Required
????

### New language strings
YES
Please help to optimize description.

### CODE REVIEW

File: components/com_wrapper/views/wrapper/tmpl/default.php [hunk 1]
```diff
@@ -9,8 +9,27 @@
...
 
+$isHtml5     = $this->params->get('mode', 'legacy') === 'html5';
+$width       = $this->escape($this->params->get('width', '100%'));
+$height      = $this->escape($this->params->get('height', '500')) . ($isHtml5 ? 'px' : '');
+$scrolling   = $this->escape($this->params->get('scrolling', 'auto'));
+$frameborder = $this->escape($this->params->get('frameborder', 1));
+
+if ($isHtml5)
+{
+
```
Comment (Quy) [inline components/com_wrapper/views/wrapper/tmpl/default.php]: Remove blank line.

File: components/com_wrapper/views/wrapper/tmpl/default.php [hunk 2]
```diff
@@ -9,8 +9,27 @@
...
+	if ($scrolling !== 'auto')
+	{
+		$scrolling = ($scrolling === 'no' ? 'hidden' : 'scroll');
```
Comment (Quy) [inline components/com_wrapper/views/wrapper/tmpl/default.php]: `()` not necessary. Remove.

File: components/com_wrapper/views/wrapper/tmpl/default.php [hunk 3]
```diff
@@ -9,8 +9,27 @@
...
+	}
+
+	$frameborder = (!$frameborder ? 'none' : $frameborder . 'px solid #000');
```
Comment (Quy) [inline components/com_wrapper/views/wrapper/tmpl/default.php]: `()` not necessary. Remove.

File: administrator/language/en-GB/en-GB.com_wrapper.ini [hunk 4]
```diff
@@ -13,10 +13,14 @@ COM_WRAPPER_FIELD_HEIGHT_LABEL="Height"
 COM_WRAPPER_FIELD_HEIGHTAUTO_DESC="If height is set to auto, the height will automatically be set to the size of the external page. This will only work for pages on your own domain. If you see a JavaScript error, make sure this parameter is disabled. This will break XHTML compatibility for this page."
 COM_WRAPPER_FIELD_HEIGHTAUTO_LABEL="Auto Height"
 COM_WRAPPER_FIELD_LABEL_SCROLLBARSPARAMS="Scroll Bar Parameters"
+COM_WRAPPER_FIELD_MODE_LABEL="Mode"
+COM_WRAPPER_FIELD_MODE_DESCRIPTION="HTML5 mode uses CSS to style the IFRAME tag. Legacy mode uses HTML 4.01 IFRAME attributes and is backwards compatible with old template overrides."
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_wrapper.ini]: Change to `COM_WRAPPER_FIELD_MODE_DESC`

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 47a52d60a50d231015a5192a81b922fa08c43e5b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19985">issues.joomla.org/tracker/joomla-cms/19985</a>.</sub>

**Comment by BaleshSrle** [comment]
You managed to work all major kinks, but there is an issue with scrolling (in HTML4) or overflow (in HTML5) in components/com_wrapper/views/wrapper/tmpl/default.xml

More about this you can see @ https://www.w3schools.com/cssref/pr_pos_overflow.asp (_auto_, _scroll_ & _hidden_ are the main options).

In CS there is no need to have line _$frameborder = !$frameborder ? 'none' : $frameborder . 'px solid #000';_. I tried to add something similar (in default.php and default.xml), but after uploading that file and refreshing my web site, I discovered an error and  I left values 1 and 0 in frameborder field (and I can inform you that works).

**Comment by ReLater** [comment]
`border: none;` is correct and v**alid** CSS. So I don't see any problems to use it.

`border: 1;` is **invalid** CSS. If you see an iframe border in your browser the reason is that the browser that you use ignores this invalid setting and falls back to it's own default CSS rules (if there are no other overruling ones in other CSS files). (EdIT: Or your browser is a nice one ;-) )



Only `border: 1px;` doesn't show a frameborder in current Joomla staging and that's the criterion for this pr. "Be B\C". 

So use `border: 1px solid #000;` to display a border like with frameborder="1" in Legacy mode.

### `border: 1px;` (HTML 5)


### `border: 1px solid #000;` (HTML 5)


### `frameborder="1"` (Legacy)


> (auto, scroll & hidden are the main options).

And these three overflow values are used in HTML5 mode and ONLY in HTML5 mode. So, I don't understand what you're talking about.

In Legacy mode nothing has changed an scrolling is yes|no|auto like before.

**Comment by ReLater** [comment]
I'm closing here until I've found a solution for `Scroll Bars` in HTML5 mode.
It looks like most browsers don't support CSS rule `overflow` for tag IFRAME.

**Comment by ReLater** [comment]
Reopened after additional commit.
I had to add a fallback for setting `Scroll Bars` .

**Pull request closed without merging**